lib/phonelib/phone_analyzer.rb

```diff
@@ -102,11 +102,11 @@ def parse_single_country(e164, data)
     # ==== Attributes
     #
     # * +phone+ - phone number for parsing
-    def detect_and_parse(phone)
+    def detect_and_parse(phone, country = nil)
       result = {}
       Phonelib.phone_data.each do |key, data|
         parsed = parse_single_country(phone, data)
-        if double_prefix_allowed?(data, phone, parsed && parsed[key])
+        if (!Phonelib.strict_double_prefix_check || key == country) && double_prefix_allowed?(data, phone, parsed && parsed[key])
```

Comment:
Wouldn't it make sense to have the `Phonelib.strict_double_prefix_check` verification on the `double_prefix_allowed?` predicate instead?

Reply:
I agree that this logic should be inside `double_prefix_allowed` but that would mean I have to pass as argument both the key and the country. In my opinion, it would make the function much less readable.